/****************************************************************************
 *
 * Copyright 2018 Samsung Electronics All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific
 * language governing permissions and limitations under the License.
 *
 ****************************************************************************/

//***************************************************************************
// Included Files
//***************************************************************************
syntax = "proto3";

package tizenrt;

message MessageDouble {
	double data = 1;
}

message MessageFloat {
	float data = 1;
}

message MessageInt32 {
	int32 data = 1;
}

message MessageInt64 {
	int64 data = 1;
}

message MessageUint32 {
	uint32 data = 1;
}

message MessageUint64 {
	uint64 data = 1;
}

message MessageSint32 {
	sint32 data = 1;
}

message MessageSint64 {
	sint64 data = 1;
}

message MessageFixed32 {
	fixed32 data = 1;
}

message MessageFixed64 {
	fixed64 data = 1;
}

message MessageSfixed32 {
	sfixed32 data = 1;
}

message MessageSfixed64 {
	sfixed64 data = 1;
}

message MessageBool {
	bool data = 1;
}

message MessageString {
	string data = 1;
}

message MessageBytes {
	bytes data = 1;
}

message MessageEnum {
  enum MessageType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }
  MessageType data = 1;
}

message MessageNested {
	int32 data = 1;
	message MessageInner {
		sint32 data = 1;
	}
	MessageInner inner = 2;
}
